src/ui/marker.js

```diff
@@ -444,35 +448,56 @@ export default class Marker extends Evented {
         return this;
     }
 
-    _updateOcclusion() {
-        if (!this._occlusionTimer) {
-            this._occlusionTimer = setTimeout(this._onOcclusionTimer.bind(this), 60);
-        }
-    }
+    _evaluateOpacity(requestAnimationFrame: boolean = false) {
+        const position = this._pos ? this._pos.sub(this._transformedOffset()) : null;
 
-    _clearOcclusionTimer() {
-        if (this._occlusionTimer) {
-            clearTimeout(this._occlusionTimer);
-            this._occlusionTimer = null;
+        if (!this._withinScreenBounds(position)) {
+            this._fadeTimer = null;
```

Comment:
Do we want to clear the timer here instead of unsetting it?

Reply:
Addressed in https://github.com/mapbox/mapbox-gl-js/pull/10564/commits/5e436a564da2c072453e1b403b755d9488d0227b